executors/cpp/message_fmt2.cpp

```diff
@@ -0,0 +1,180 @@
+ * testing message format 2 in C++
+ */
+
+#include <unicode/utypes.h>
+
+#include <json-c/json.h>
+
+// This API was added in ICU75.1
+#if U_ICU_VERSION_MAJOR_NUM >= 75
+#include <unicode/locid.h>
+#include <unicode/messageformat2.h>
+#include <unicode/messageformat2_arguments.h>
+#include <unicode/messageformat2_data_model.h>
+#include <unicode/messageformat2_function_registry.h>
+#include <unicode/messageformat2_formattable.h>
+#include <unicode/msgfmt.h>
+#include <unicode/unistr.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+#include <string>
+#include <cstring>
+
+using icu::Locale;
+using icu::message2::MessageFormatter;
+using icu::message2::MessageArguments;
+using icu::message2::Formattable;
+
+using icu::UnicodeString;
+
+using std::string;
+
+/* Based on this test file:
+ * https://github.com/unicode-org/icu/blob/main/icu4c/source/test/intltest/messageformat2test.cpp
+ */
+const string TestMessageFormat2(json_object *json_in) {
+  UErrorCode errorCode = U_ZERO_ERROR;
+
+  // label information
+  json_object *label_obj = json_object_object_get(json_in, "label");
+  string label_string = json_object_get_string(label_obj);
+
+  json_object *locale_label_obj = json_object_object_get(json_in, "locale");
+  string locale_string = "und";
+  if (locale_label_obj) {
+    locale_string = json_object_get_string(locale_label_obj);
+  }
+
+  const Locale displayLocale(locale_string.c_str());
+
+  string src_string;
+  json_object *src_obj = json_object_object_get(json_in, "src");
+  UnicodeString u_src;
+  if (src_obj) {
+    src_string = json_object_get_string(src_obj);
+    u_src = src_string.c_str();
+  }
+
+  string test_description_string;
+  json_object *test_description_obj =
+      json_object_object_get(json_in, "test_description");
+  if (test_description_obj) {
+    test_description_string = json_object_get_string(test_description_obj);
+  }
+
+  json_object* param_list_obj = json_object_object_get(json_in, "params");
+
+  string params_name;
+  UnicodeString u_params_name;
+  string params_value;
+  UnicodeString u_params_value;
+
+  // Get all the argements from test params
+  std::map<UnicodeString, Formattable> argsBuilder;
+
+  if (param_list_obj) {
+    array_list* param_list = json_object_get_array(param_list_obj);
+
+    // For each item in the params list, get the name and value.
+    // Add to the builder.
+    int params_length = json_object_array_length(param_list_obj);
+
+    // Construct the list of Unicode Strings
+    for (int i = 0; i < params_length; i++) {
+      json_object* param_obj = json_object_array_get_idx(param_list_obj, i);
+
+      json_object *params_name_obj = json_object_object_get(param_obj, "name");
+      if (params_name_obj) {
+        params_name = json_object_get_string(params_name_obj);
+        u_params_name = params_name.c_str();
+      }
+      json_object *params_value_obj =
+          json_object_object_get(param_obj, "value");
+      if (params_value_obj) {
```

Comment:
suggestion: if you move `json_object *params_value_obj = ...` up to where `json_object *params_name_obj = ...` is, then you can combine `if (params_name_obj) {` and `if (params_value_obj) {`, and then add an `else` branch saying `continue`.

Right now, if either the name or the value is empty, then it seems meaningless and potentially confusing/incorrect to try to add the name+value pair to the args builder.